### PR TITLE
Upgrading the version from 8.0.0 to 8.0.3

### PR DESCRIPTION
**Pull request by Sfurti-yb**
To support entity framework with YugabyteDB smart driver, the driver needed to be upgraded as the postgres adapter for EF expects version 8.0.3 of the Npgsql driver.

Changes made:
Rebased the smart driver changes on top of v8.0.3 of upstream driver
Fixed issues with nodeup behaviour of fallback and added tests
Added flags to suppress Discard sqls (ported changes from v6.0.10 of the driver)

Tests done:
Uniform load balance tests (`LoadBalanceTests.cs`)
Topology Aware Tests (`FallBackTopologyTests.cs` and `FallbackTopologyTestsExtended.cs`)

### CODE REVIEW

File: NuGet.config [hunk 1]
```diff
@@ -2,6 +2,7 @@
 <configuration>
 
     <packageSources>
+        <add key="Package source 1" value="/Users/sfurtisarah/work/LocalNuget" />
```
Comment (ashetkar) [inline NuGet.config]: Why is this needed? In either case, we should not use a hard-coded local path.

File: README.md [hunk 2]
```diff
@@ -51,3 +51,57 @@ await using (var reader = await cmd.ExecuteReaderAsync())
...
+
+### Usage
+
+Load balancing connection properties:
+
+The following connection properties are added to enable load balancing:
+
+* Load Balance Hosts - enable cluster-aware load balancing by setting this property to True; disabled by default.
+* Topology Keys - provide comma-separated geo-location values to enable topology-aware load balancing. Geo-locations can be provided as cloud.region.zone.
+* YB Servers Refresh Interval - The list of servers, to balance the connection load on, are refreshed periodically every 5 minutes by default. This time can be regulated by this property.
```
Comment (ashetkar) [inline README.md:83]: We should also mention other connection properties like fallback-to-topology-keys-only, failed-host-reconnect-delay-secs

File: NuGet.config (hunk 1 above)
Comment (Sfurti-yb) [reply NuGet.config]: It wasn't needed, it got committed from my local config. Removed it now.

File: README.md (hunk 2 above)
Comment (Sfurti-yb) [reply README.md:83]: Mentioned the other flags as well.

**Review by ashetkar — APPROVED** [review]
